client/client.go

```diff
@@ -218,12 +221,39 @@ func (cli *Client) ClientVersion() string {
 // UpdateClientVersion updates the version string associated with this
 // instance of the Client. This operation doesn't acquire a mutex.
 func (cli *Client) UpdateClientVersion(v string) {
+	// If no version specified, negotiate with the server
+	if v == "" {
+		cli.DowngradeClientVersion()
+	}
```

Comment:
Indeed there are no other references to UpdateClientVersion that I can find, however removing it would:
* potentially break existing consumers of the client library
* put update logic into the new negotiate function (although, it is just updating a field)